python_modules/dagster/dagster/_core/definitions/data_time.py

```diff
@@ -57,31 +56,13 @@ def _calculate_data_time_partitioned(
         beginning of that time window, or all data up until the end of the first filled time window
         in the total set, whichever is less.
         """
-        from dagster._core.storage.partition_status_cache import (
-            get_and_update_asset_status_cache_value,
+        # the total set of materialized partitions
+        partition_subset = partitions_def.empty_subset().with_partition_keys(
```

Comment:
In the unpartitioned asset case, this could be called once for each upstream partition right? Can we cache the data in this case to avoid recalculating?

Reply:
Are you thinking about the case where an unpartitioned is downstream of a time-partitioned asset? In this codepath, that will still only result in a single call to `_calculate_data_time_partitioned`, as the data time of a partitioned asset is a property of the entire asset, rather than a property of each partition of that asset.

So in general, in this part of the world, the number of queries does not scale with the number of partitions in the graph